### ISSUE TITLE
GraphQL - Umbraco Heartcore

### ISSUE THREAD

**Issue by Hanskrogh** [umbraco/Umbraco.Cloud.Issues#548]
### Issue description

We're experiencing the following issue with our Umbraco heartcore solution:

Everytime we do an update to an document/data type, the graphql for that specific document/element will be down for a certain amount of time. When we do this on simple documents/datatypes with not a lot of references, it might take 1-5 mins, which is fine, but the issue comes when we do it to a large document/data type, then the graphql can be out of functioning for multiple hours before it is up to date.

We have a really large nested content element named  "Component picker" which is use many places on in our solution. When updating this, it takes serval minutes just to save it and once it is saved, the graphql will return an empty array, even though there is content inside it. Then we wait for some hours and it will be back functioning as it is supposed to.

Is this something you have experienced before?

**Comment by Hanskrogh** [comment]
It is not once the content is saved, it is once the document/data-type is saved, all content related to that document/data-type returns null / is empty.

**Comment by sitereactor** [comment]
Ah, okay. Thanks for clearing that up.

When anything related to the schema is updated, we push all content into the Content Delivery Platform, which is where the GraphQL API lives. This means the entire content cache is refreshed. That does take a little bit of time, but ideally not more than a few minutes max.
We have seen some cases where the GraphQL content is stale (meaning that it returns old content) for a longer period of time. There are a few things we are working on to improve this. I haven't seen cases where the content is null/empty, but we can try to add some test cases to see if we can replicate the issue you are seeing.

Hopefully I can provide an update next week.

**Comment by Hanskrogh** [comment]
Would you like me to make a short screen record of me triggering the error and showing the result?

**Comment by sitereactor** [comment]
Yes, that would be great. You can email it to me if you prefer to not post it here (my email is first name at umbraco dot com).
